src/sentry/models/organizationmapping.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import BoundedBigIntegerField, Model
+from sentry.db.models.base import control_silo_only_model
+
+
+@control_silo_only_model
+class OrganizationMapping(Model):
+    """
+    This model is used to:
+    * Map org slugs to a specific organization and region
+    * Safely reserve organization slugs via an eventually consistent cross silo workflow
+    """
+
+    __include_in_export__ = True
+
+    organization_id = BoundedBigIntegerField(db_index=True)
+    slug = models.SlugField(unique=True)
+    # TODO(hybrid-cloud): Name is currently blank for all records. Updating an org name should happen for all applicable slugs.
+    name = models.CharField(max_length=64)
+    created = models.DateTimeField(default=timezone.now)
+    stripe_id = models.CharField(max_length=255, db_index=True, null=True)
```

Comment:
I don't love that we're including `stripe_id` here at the sentry level. ST and self-hosted customers don't care about the stripe at all, it's billing specific which we usually try to keep contained in getsentry. Is there a more generic way we might be able to handle this?

Reply:
That would be better. I'd prefer that we keep billing specific attributes out of sentry in general, although I understand that might be difficult. I think it's worth at least considering whether there's a way to keep this logic contained in getsentry though.